### PR TITLE
Only send daily deploy for certain environments

### PR DESCRIPTION
**Pull request by sheelio**
@dannyroberts 
Feedback about the daily deploy emails, so changing them to only go eNikshay / ICDS / production. 

Does this work?

### CODE REVIEW

**Comment by dannyroberts** [comment]
I don't think this approach is quite right.

These emails used to go to commcarehq-ops@, so rather than telling it not to send (which I don't think this PR _quite_ does, but that's the obvious intention), I'd say just change the email on those environments back to the old one. Does that work?

**Comment by sheelio** [comment]
These emails actually still go to both (ops and daily announce).  This just turns it off for daily-announce but keeps the ops behaviour the same.

**Comment by dannyroberts** [comment]
Ah, I see. In that case, I'd suggest making a change like https://github.com/dimagi/commcare-hq/pull/18497 and then just setting the value to `null` in the yml config files for the envs. As you have it now, the `settings.py` default (an `@example.com`) address will bleed through, and that's not really what we want.

File: ansible/group_vars/all.yml [hunk 1]
```diff
@@ -146,4 +146,3 @@ report_builder_add_on_email: sales@dimagi.com
 eula_change_email: eula-notifications@dimagi.com
 contact_email: info@dimagi.com
 soft_assert_email: commcarehq-ops+soft_asserts@dimagi.com
-daily_deploy_email: tech-announce-daily@dimagi.com
```
Comment (dannyroberts) [inline ansible/group_vars/all.yml]: I believe you can override this directly in the *`_public.yml` files, so you can just set it to `null` here and add `daily_deploy_email: tech-announce-daily@dimagi.com` in those three other files without changing `localsettings.py.j2`.

**Comment by sheelio** [comment]
@dannyroberts 
Thanks - updated

**Pull request merged**